core/src/main/java/org/ehcache/core/EhcacheManager.java

```diff
@@ -63,76 +110,548 @@ public EhcacheManager(Configuration config, Collection<Service> services) {
+          serviceLocator.stopAllServices();
+        } catch (Exception exceptionStoppingServices) {
+          this.getLogger().error("Stopping services failed due to ", exceptionStoppingServices);
         }
+        throw e;
       }
-      try {
-        LocalPersistenceService.PersistenceSpaceIdentifier space = persistenceService.getOrCreatePersistenceSpace(alias);
-        adjustedServiceConfigs.add(space);
-      } catch (CachePersistenceException cpex) {
-        throw new RuntimeException("Unable to create persistence space for cache " + alias, cpex);
+    } catch (Exception e) {
+      throw st.failed(e);
+    }
+    st.succeeded();
+  }
+
+  @Override
+  public Status getStatus() {
+    return statusTransitioner.currentStatus();
+  }
+
+  @Override
+  public void close() {
+    final StatusTransitioner.Transition st = statusTransitioner.close();
+
+    Exception firstException = null;
+    try {
+      for (String alias : caches.keySet()) {
+        try {
+          removeCache(alias, false);
+        } catch (Exception e) {
+          if(firstException == null) {
+            firstException = e;
+          } else {
+            this.getLogger().error("Cache '{}' could not be removed due to ", alias, e);
+          }
+        }
+      }
+
+      serviceLocator.stopAllServices();
+    } catch (Exception e) {
+      if(firstException == null) {
+        firstException = e;
+      }
+    }
+    if(firstException != null) {
+      throw st.failed(firstException);
+    }
+    st.succeeded();
+  }
+
+  @Override
+  public RuntimeConfiguration getRuntimeConfiguration() {
+    return configuration;
+  }
+
+  /**
+   * Removes and closes a cache without performing {@link CacheManagerListener#cacheRemoved(String, Cache)}
+   * notifications.
+   *
+   * @param alias the alias of the cache to remove
+   */
+  protected void removeAndCloseWithoutNotice(final String alias) {
+    final CacheHolder cacheHolder = caches.remove(alias);
+    if(cacheHolder != null) {
+      final Ehcache<?, ?> ehcache = cacheHolder.retrieve(cacheHolder.keyType, cacheHolder.valueType);
+      if(ehcache.getStatus() == Status.AVAILABLE) {
+        ehcache.close();
       }
     }
+  }
 
-    return super.getStore(alias, config, keyType, valueType, adjustedServiceConfigs, lifeCycledList);
+  @Override
+  public void destroyCache(final String alias) throws CachePersistenceException {
+    if (alias == null) {
+      throw new NullPointerException("Alias cannot be null");
+    }
+    this.getLogger().info("Destroying Cache '{}' in {}.", alias, simpleName);
+    removeAndCloseWithoutNotice(alias);
+    destroyPersistenceSpace(alias);
+    this.getLogger().info("Cache '{}' is successfully destroyed in {}.", alias, simpleName);
+  }
+
+  private void destroyPersistenceSpace(String alias) throws CachePersistenceException {
+    Collection<PersistableResourceService> services = serviceLocator.getServicesOfType(PersistableResourceService.class);
```

Comment:
Are there any concerns about ordering of destruction of multiple persistent services?